pkg/solana/logpoller/poller.go

```diff
@@ -0,0 +1,121 @@
+package logpoller
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+type EventSaver interface {
+	SaveEvent(evt ProgramEvent) error
+}
+
+type Service struct {
+	// dependencies
+	lggr  logger.Logger
+	saver EventSaver
+
+	// internal
+	loader         *EncodedLogCollector
+	mu             sync.RWMutex
+	discriminators map[string]struct{}
+	chSave         chan ProgramEvent
+
+	// service state management
+	services.Service
+	engine *services.Engine
+}
+
+func New(client RPCClient, lggr logger.Logger, saver EventSaver) *Service {
+	p := &Service{
+		saver:          saver,
+		discriminators: make(map[string]struct{}),
+		chSave:         make(chan ProgramEvent),
+	}
+
+	p.Service, p.engine = services.Config{
+		Name: "LogPollerService",
+		NewSubServices: func(lggr logger.Logger) []services.Service {
+			p.loader = NewEncodedLogCollector(client, p, lggr)
+
+			return []services.Service{p.loader}
+		},
+		Start: p.start,
+		Close: p.close,
+	}.NewServiceEngine(lggr)
+	p.lggr = p.engine.SugaredLogger
+
+	return p
+}
+
+func (p *Service) AddFilter(name string) error {
```

Comment:
This looks like `RegisterFilter()` we have that implemented and is in review.

Reply:
The interface as well as all of the important data structures for LogPoller is fully specified in the design doc, that should be enough for QueryKey to be implemented so I'd recommend starting there, and if there is anything missing we can discuss.